instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/net/internal/InternalNetServerAttributesExtractor.java

```diff
@@ -10,37 +10,35 @@
 import io.opentelemetry.api.common.AttributesBuilder;
 import io.opentelemetry.instrumentation.api.instrumenter.net.NetServerAttributesGetter;
 import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
-import java.util.Locale;
 import java.util.function.BiPredicate;
 
 /**
  * This class is internal and is hence not for public use. Its APIs are unstable and can change at
  * any time.
  */
-public final class InternalNetServerAttributesExtractor<REQUEST> {
+public final class InternalNetServerAttributesExtractor<REQUEST, RESPONSE> {
 
-  private final NetServerAttributesGetter<REQUEST> getter;
+  private final NetServerAttributesGetter<REQUEST, RESPONSE> getter;
   private final BiPredicate<Integer, REQUEST> captureHostPortCondition;
   private final FallbackNamePortGetter<REQUEST> fallbackNamePortGetter;
+  private final boolean emitOldHttpAttributes;
 
   public InternalNetServerAttributesExtractor(
-      NetServerAttributesGetter<REQUEST> getter,
+      NetServerAttributesGetter<REQUEST, RESPONSE> getter,
       BiPredicate<Integer, REQUEST> captureHostPortCondition,
-      FallbackNamePortGetter<REQUEST> fallbackNamePortGetter) {
+      FallbackNamePortGetter<REQUEST> fallbackNamePortGetter,
+      boolean emitOldHttpAttributes) {
```

Comment:
doesn't this whole class only apply when emitOldHttpAttributes = true?

Reply:
Not yet -- but soon will be.